src/DynamoManipulation/NodeManipulator.cs

```diff
@@ -494,19 +494,31 @@ private RenderPackageCache GenerateRenderPackages()
                 return packages;
             }
 
-            // This check is required as for some reason LibG fails to load, geometry nodes are null
-            // and we must return immediately before proceeding with further calls to ProtoGeometry
+            // This check is required if for some reason LibG fails to load, geometry nodes are null
+            // and we must return immediately before proceeding with further calls to ProtoGeometry.
+            // It's possible the the node's return value is not null, but libG was still not loaded correclty.
+            // For example we could be returning a function, and checking for null or function is not sufficent here
+            // as direct manipulators should be created for function outputs. Additionally, checking for Function does not 
```

Comment:
`direct manipulators should be created for function outputs` - I'm not sure I agree with this statement. Can you explain why you think manipulators should be created for function outputs?

Reply:
@aparajit-pratap let me do some more testing with the function check, I may have mis understood when a manipulator should be created, in which case the function || null check may work.